### ISSUE TITLE
Has there been any desire to submit pull requests to make QuantLibExt part of Quantlib itself?

### ISSUE THREAD

**Issue by mrslezak** [OpenSourceRisk/Engine#83]
I'm just wondering why, I know it's not required by the BSD license, but is there a reason for leaving the QuantLibExt extensions outside of the Quantlib library itself?  If it was pushed to the QuantLib library, there would be many more "eyes" on it from the open source community, and therefore likely more development upon its work.  I understand it is the work of the company in support of Open Source Risk, but this would potentially simplify the development ahead for both ORE and QuantLib.

**Comment by mrslezak** [comment]
Okay I started the port of QuantExt and resolved the errors on VS2017 Windows 10 with Boost 1.77.0 and QuantLib 1.23, using the Windows 8.1SDK and haven't finished the testsuite.  Once the testsuite is done, it should be about ready to merge into QuantLib.  If anyone wants to help with this, my repo is here: https://github.com/mrslezak/Engine check the 1st commit there were various issues with the formatting using the code as is posted here on Windows 10, but now it compiles fine.  The offline installer for VS2017 Community was failing for me on download so I ended up using this ISO and then adding the C++ needed features afterwards: https://drive.google.com/file/d/1b5lqWOE421vDw-wDPxBBxixEcienyhx2

**Comment by pcaspers** [comment]
We use our own fork of QuantLib (as a git submodule of this repo) with a few extensions and patches. The date generation rule you mention is defined here

https://github.com/OpenSourceRisk/QuantLib/blob/77ecf0a9d573234f5db7a02f75a3caa65ab57662/ql/time/dategenerationrule.hpp

We strive to get our patches into the main QuantLib repo over time, so that our submodule shows minimal and only temporary differences to the official QuantLib release.

As for integrating QuantExt into QuantLib, we are considering this. Thanks for starting the work, but this less a technical than an organisational question, so that your efforts might be futile at this point I guess.

**Comment by mrslezak** [comment]
My true interest in all of this is to actually make a better Python interface to all the libraries.  As the SWIG interface is a little archaic, I imagine that https://github.com/StatisKit/AutoWIG which auto-wraps to a Boost.Python interface would be cleaner.  Not to say it's auto-wrapping will work perfectly without any tweaks, but it would help port it over.  It also can wrap to PyBind11 but I'm not so sure that would work without removing the Boost dependencies from QuantLib (which hasn't been done since v1.10 here: https://github.com/haozhangphd/QuantLib-noBoost) and is unlikely to be embraced by the QuantLib project...  Since they knew of its existence, but never extended it further in some older emails on the QuantLib discussion forums.  What all has changed in v1.23 I'm not sure but that's quite a bit newer.
